### PR TITLE
Updated the Bash guide wrt introduction of bash jinja macros.

### PR DESCRIPTION
**Pull request by matejak**
- Changed the guidance to use 4-space for indent.
- Cleaned up the example where die is used.

### CODE REVIEW

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -1176,7 +1176,12 @@ fi
 
 When writing new bash remediations content, please follow the following guidelins:
 
-* Use tabs for indentation rather than spaces.
+* Use four spaces for indentation rather than tabs.
+* You can use macros from `shared/macros-bash.jinja` in the remediation content.
+If the macro is used from a nested block, use the `indent` jinja2 filter assuming the 4-space indentation.
+Typically, you want to call the macro with the intended indentation, and as `indent` doesn't indent the first line by default, you just pass the number of spaces as the only argument.
+
+See the remediation for rule `ensure_fedora_gpgkey_installed` for reference.
```
Comment (jan-cerny) [inline docs/manual/developer_guide.adoc]: This has destroyed the rest of the bullet list. Please fix this to make the list displayed as a list.

File: linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh [hunk 2]
```diff
@@ -20,22 +20,22 @@ readonly REDHAT_RELEASE_KEY="/etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-${fedora_versio
...
+    # Verify /etc/pki/rpm-gpg directory permissions are safe
+    if [ "${RPM_GPG_DIR_PERMS}" -le "755" ]; then
+        # If they are safe, try to obtain fingerprints from the key file
+        # (to ensure there won't be e.g. CRC error).
+        readarray -t GPG_OUT < <(gpg --show-keys --with-fingerprint --with-colons "${REDHAT_RELEASE_KEY}" | grep '^fpr' | cut -d ":" -f 10)
+        GPG_RESULT=$?
+        # No CRC error, safe to proceed
+        if [ "${GPG_RESULT}" -eq "0" ]; then
+            echo "${GPG_OUT}" | grep -vE "${FEDORA_RELEASE_FINGERPRINT}" || {
+            # If file doesn't contains any keys with unknown fingerprint, import it
```
Comment (jan-cerny) [inline linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh]: should be "doesn't contain"

**Comment by matejak** [comment]
Good catch, you can recheck.

**Comment by jan-cerny** [comment]
@matejak I confirm the list is now a list. Waiting for CI, but LGTM.

**Pull request merged**